RELEASE.md

```diff
@@ -0,0 +1,17 @@
+# Publishing a New Release
+*This page contains instructions for admins of this project to release a new version.*
+
+## Run Makefile
+Run `$ make publish VERSION_NUMBER` at the root directory of needle (where the `Makefile` is located)
+
+For example:
+```
+$ make publish 0.13.0
+```
+
+This runs the steps specified in the `Makefile`.
+
+The `Makefile` does not support failure recovery yet. If a certain step fails, you might want to figure out which step failed, and manually run the remaining steps.
+
+## Create a new Github release
```

Comment:
Is this for external devs? Do we want to allow them to publish releases?  @rudro

Reply:
I don't think we want to allow allow non-admins to make releases.
As written in the first paragraph of this doc, it's intended for collaborating admins like myself.

Reply:
My point is, do we want to share this info on readme whose target audience is external devs?  I'll leave this up to @rudro.

Reply:
I think it's fine to be public. It's relatively standard steps.  True it's only useful to the 2-3 people with admin access to this repo, but don't think it should confuse anyone.